### PR TITLE
Fix guests never thinking that they're being watched

### PR DESCRIPTION
**Pull request by rctdude2**
This actually fixes two issues:

1. Guests never think that they're being watched
2. The animation for the guest overview tab updates twice per call (especially obvious on guests with a balloon)

The way it is before this PR, both `w->var_496` and `w->highlighted_item` are in a union. Since the animation is updated first, this means `w->highlighted_item` gets reset to 0 when it hits 24 (old line 1029), AND `w->var_496` gets updated twice per call (old lines 1028 and 1051), making the tab animation run too fast and preventing the timer from ever reaching the minimum time of 3840.

This PR fixes both by just adding a separate variable outside of the union for the timer.

### CODE REVIEW

**Comment by rctdude2** [comment]
I think the lint commit message failed because I committed the changelog update too quickly, it's complaining about an invalid revision range? I know the messages are <80 characters.

**Comment by IntelOrca** [comment]
While this fixes the bug, it raises the question that this will desync multiplayer. For now, can you only execute this code if in single player?

**Comment by rctdude2** [comment]
Should I increment network stream version as well?

**Comment by IntelOrca** [comment]
> Should I increment network stream version as well?

If you are saying that it never triggered prior to this PR, then I think you are safe.

**Comment by rctdude2** [comment]
This has been broken since at least late 2020 as far as I can tell.

(look at the peep animation in the tab, it "pops" every couple of seconds)
https://youtu.be/X9ooa7lPlO4?t=10294

File: src/openrct2/interface/Window_internal.h [hunk 1]
```diff
@@ -94,6 +94,7 @@ struct rct_window
         const scenario_index_entry* highlighted_scenario;
         uint16_t var_496;
     };
+    uint32_t guestWatchTimer; // Timer for "someone is watching me" thought
```
Comment (duncanspumpkin) [inline src/openrct2/interface/Window_internal.h:97]: Thus shouldn't become part of this struct it's specific to the peep window. The peep window will need to be refactored to use the new style first

**Comment by duncanspumpkin** [comment]
> While this fixes the bug, it raises the question that this will desync multiplayer. For now, can you only execute this code if in single player?

It has been protected from that since multiplayer was added

File: src/openrct2-ui/windows/Guest.cpp [hunk 2]
```diff
@@ -1045,18 +1046,18 @@ void WindowGuestOverviewUpdate(rct_window* w)
...
-        w->highlighted_item &= 0xFFFF0000;
-    else
-        w->highlighted_item++;
-
     // Disable peep watching thought for multiplayer as it's client specific
     if (network_get_mode() == NETWORK_MODE_NONE)
     {
-        // Create the "I have the strangest feeling I am being watched thought"
-        if ((w->highlighted_item & 0xFFFF) >= 3840)
+        if (w->guestWatchTimer == 0xFFFF)
```
Comment (CookiePLMonster) [inline src/openrct2-ui/windows/Guest.cpp:1052]: Unlike signed integers, unsigned integer overflow is well defined - if this timer was a `uint16_t` the code could safely let it wrap around to 0 without the need for this check.

**Comment by Gymnasiast** [comment]
Superseded by https://github.com/OpenRCT2/OpenRCT2/pull/17471.

**Pull request closed without merging**